documentation/MIGRATION-guide-for-next-generation-management-libraries.md

```diff
@@ -87,61 +95,63 @@ getInstanceView(
+
 ```typescript
-await client.availabilitySets.list(this.resourceName).then(
-    response => handle(response)
-)
+await client.availabilitySets.list(this.resourceName).then((response) => handle(response));
 ```
-now you will get a iterator, and you need to do the iteration to get the result. 
+
+now you will get a iterator, and you need to do the iteration to get the result.
+
 ```typescript
 const result = client.availabilitySets.list(this.resourceName);
 for await (const item of result) {
-    handle(item);
+  handle(item);
 }
 ```
+
 The newly added `PagedAsyncIterableIterator` also allows you to get these items by page if the List operation result is too long.
+
 ```typescript
 const result = client.availabilitySets.list(this.resourceName);
-for await (const item of result.byPage({maxPageSize:2})) {
-    handle(item);
+for await (const item of result.byPage({ maxPageSize: 2 })) {
+  handle(item);
 }
 ```
+
 Refer to [@azure/core-paging](https://www.npmjs.com/package/@azure/core-paging) for more details.
 
 ## Long Running Operations
 
-Javascript/Typescript SDK provide two operations with the same signature for Long Running Operations for our customers. One operation returns a LROPoller whose name starts with `begin` prefix, the other will poll until finished and return the final result to our customer that shares the same name with the rest api operationId.  
-In next-generation, we keep this feature except 
-1. we change those operations names with a `begin` prefix and `AndWait` suffix which will poll until finshed and return the result directly.  
-1. we change the poller type of the response
+Many operations could take a long time before sending the desired response. The SDK provides two types of methods to interact with such operations. First type is a method that simply returns the result after the operation finishes processing and those methods' names correspond to the name of the API they call. One issue with these methods is that they do not provide a way to check on the current status of the operation or to access any partial results computed so far. The next-generation SDK provides similar methods but with a slightly different name, it has the prefix `begin` and the postfix `AndWait`. The other type of methods is one that returns a poller object which gives you access to the underlying state of the operation. Previously, these methods returned an object of type [`LROPoller`](https://github.com/Azure/ms-rest-azure-js/blob/a9cee4480a8710d5c81890ae7cb7a1dea559ec65/lib/lroPoller.ts#L13), a class that is exported by `@azure/ms-rest-azure-js` but the new operations return an object that implements the [`PollerLike`](https://github.com/Azure/azure-sdk-for-js/blob/35f86a68def5022d4c03a32d4c8c0362a0158772/sdk/core/core-lro/src/poller.ts#L50) interface instead which is exported by `@azure/core-lro`. The name of those methods did not change, both versions use the prefix `begin` with no postfix.
```

Comment:
Do we really have to mention what's the drawback of beginXXXAndWait operation in here ?  I think we can have a independent section that introduces the pros and cons for both type of operations. @nickzhums  what do you think ?

Reply:
This is a migration guide and I do not think it is meant to be a learning resource about LROs in general. So I think having it in a separate section is overkill. I briefly mentioned the disadvantage here just in case the reader is not familiar with them and they're already using the poller-returning one.

Reply:
i think @deyaaeldeen 's suggestion makes sense. For general LRO patern, we can add it to the quick-start / overview docs or sth like that

Reply:
What I am trying to suggest is that we don't need to mention neither the issue with beginXXXAndWait operation or the benefit with Poller. Customers who are familier with track1 SDK, they are sure to know which one is what they wanted. Besides, All the benefit/feature of the new poller has already been covered in `## Why switch to the next-generation` section.